packages/react/src/hooks/user.ts

```diff
@@ -680,3 +683,59 @@ export const useConfigMfa: ConfigMfaHook = () => {
     error
   }
 }
+
+/**
+ * Decode the current decoded access token (JWT), or return `null` if the user is not authenticated (no token)
+ * @see {@link https://hasura.io/docs/latest/graphql/core/auth/authentication/jwt/|Hasura documentation}
+ * @example
+ * ```ts
+ * import { useDecodedAccessToken } from '@nhost/react'
+ * const Component = () => {
+ *    const decodedToken = useDecodedAccessToken()
+ *    return <div>Decoded access token: {JSON.stringify(decodedToken)}</div>
+ * }
+ * ```
+ */
+export const useDecodedAccessToken = () => {
+  const jwt = useAccessToken()
+  return useMemo(() => (jwt ? jwt_decode<JWTClaims>(jwt) : null), [jwt])
+}
+
+/**
+ * Decode the Hasura claims from the current access token (JWT) located in the `https://hasura.io/jwt/claims` namespace, or return `null` if the user is not authenticated (no token)
+ * @see {@link https://hasura.io/docs/latest/graphql/core/auth/authentication/jwt/|Hasura documentation}
+ * @example
+ * ```ts
+ * import { useHasuraClaims } from '@nhost/react'
+ * const Component = () => {
+ *    const hasuraClaims = useHasuraClaims()
+ *    return <div>JWT claims in the `https://hasura.io/jwt/claims` namespace: {JSON.stringify(hasuraClaims)}</div>
+ * }
+ * ```
+ */
+export const useHasuraClaims = () => {
+  const claims = useDecodedAccessToken()
+  return useMemo(() => claims?.['https://hasura.io/jwt/claims'] || null, [claims])
+}
+
+/**
+ * Get the value of a given Hasura claim in the current access token (JWT). Returns null if the user is not authenticated, or if the claim is not in the token.
+ * Return `null` if the user is not authenticated (no token)
+ * @param name name of the variable. Automatically adds the `x-hasura-` prefix if it is missing
+ * @see {@link https://hasura.io/docs/latest/graphql/core/auth/authentication/jwt/|Hasura documentation}
+ * @example
+ * ```ts
+ * import { useHasuraClaim } from '@nhost/react'
+ * const Component = () => {
+ *    const claim = useHasuraClaim('user-id')
+ *    return <div>User id extracted from the JWT access token: {claim}</div>
+ * }
+ * ```
+ */
+export const useHasuraClaim = (name: string) => {
+  const hasuraClaims = useHasuraClaims()
+  return useMemo(
```

Comment:
`useMemo` is unnecessary here, because we would recalculate the return value either way. Either if `name` or `hasuraClaims` changes.